src/uu/sync/src/sync.rs

```diff
@@ -169,12 +175,45 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         .map(|v| v.map(ToString::to_string).collect())
         .unwrap_or_default();
 
+    if matches.is_present(options::DATA) && files.is_empty() {
+        return Err(USimpleError::new(1, "--data needs at least one argument"));
+    }
+
     for f in &files {
-        if !Path::new(&f).exists() {
-            return Err(USimpleError::new(
-                1,
-                format!("cannot stat {}: No such file or directory", f.quote()),
-            ));
+        // Use the Nix open to be able to set the NONBLOCK flags for fifo files
+        #[cfg(any(target_os = "linux", target_os = "android"))]
+        {
+            match open(Path::new(&f), OFlag::O_NONBLOCK, Mode::empty()) {
+                Ok(_) => {}
+                Err(e) => {
+                    if e == Errno::ENOENT {
```

Comment:
We should probably add automatic conversion from `nix::Errno` to `UError` like we have for `std::io::Error`. Then we don't need these manual conversions.

Reply:
Sounds like a good idea. Do you want me to open an issue as a good first bug ?

Reply:
Yeah good idea